src/pages/settings/AddSecondaryLoginPage.js

```diff
@@ -0,0 +1,156 @@
+
+            // Date of when login was validated
+            validatedDate: PropTypes.string,
+        })),
+    }),
+
+    // Route object from navigation
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            type: PropTypes.string,
+        }),
+    }),
+};
+
+const defaultProps = {
+    user: {},
+    route: {},
+};
+
+class AddSecondaryLoginPage extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            login: '',
+            password: '',
+        };
+        this.formType = props.route.params.type;
+        this.submitForm = this.submitForm.bind(this);
+        this.validateForm = this.validateForm.bind(this);
+    }
+
+    componentWillUnmount() {
+        Onyx.merge(ONYXKEYS.USER, {error: ''});
+    }
+
+    submitForm() {
+        setSecondaryLogin(this.state.login, this.state.password)
+            .then((response) => {
+                if (response.jsonCode === 200) {
+                    Navigation.navigate(ROUTES.SETTINGS_PROFILE);
+                }
+            });
+    }
+
+    // Determines whether form is valid
+    validateForm() {
+        const validationMethod = this.formType === CONST.LOGIN_TYPE.PHONE ? Str.isValidPhone : Str.isValidEmail;
+        return !this.state.password || !validationMethod(this.state.login);
+    }
+
+    render() {
+        return (
+            <ScreenWrapper>
+                <HeaderWithCloseButton
+                    title={this.formType === CONST.LOGIN_TYPE.PHONE ? 'Add Phone Number' : 'Add Email Address'}
+                    shouldShowBackButton
+                    onBackButtonPress={() => Navigation.navigate(ROUTES.SETTINGS_PROFILE)}
+                    onCloseButtonPress={Navigation.dismissModal}
+                />
+                <View style={[styles.p5, styles.flex1, styles.overflowScroll]}>
+                    <View style={styles.flexGrow1}>
+                        <Text style={[styles.mb6, styles.textP]}>
+                            {this.formType === CONST.LOGIN_TYPE.PHONE
+                                ? 'Enter your preferred phone number and password to send a validation link.'
+                                : 'Enter your preferred email address and password to send a validation link.'}
+                        </Text>
+                        <View style={styles.mb6}>
+                            <Text style={[styles.mb1, styles.formLabel]}>
+                                {this.formType === CONST.LOGIN_TYPE.PHONE ? 'Phone Number' : 'Email Address'}
+                            </Text>
+                            <TextInput
+                                style={styles.textInput}
+                                value={this.state.login}
+                                onChangeText={login => this.setState({login})}
+                                autoFocus
+                                keyboardType={this.formType === CONST.LOGIN_TYPE.PHONE ? 'phone-pad' : undefined}
```

Comment:
NAB: but maybe we should introduce constants for the various keyboardTypes? I see `phone-pad` and `numeric`